### ISSUE TITLE
Exception reported: Invalid JSON in line 1, column 16014: premature end of input

### ISSUE THREAD

**Issue by agpmilli** [ErikMinekus/sm-ripext#57]
After sending a GET request, we receive a response that contains ~25'000 characters. it comes with status code 200 but when fetching the data (`response.Data`), we obtain errors like these (it varies):

- `Exception reported: Invalid JSON in line 1, column 16014: premature end of input`
- `Exception reported: Invalid JSON in line 1, column 16016: invalid escape`

The column varies between 16014 and 16016 for this error to occur and the error message between the two above.

Is there a size limit to the JSON object? Can we bypass it?
Any other idea on why it occurs?

**Comment by agpmilli** [comment]
Just to give a bit more context. we have been using sm-ripext to read JSON from body string (Calls made using SteamWorks and body returned in string). 
It was working well when using `FromString` function. So it seems that there is not size limit there.

We felt it would be more optimal to use ripext completely, both to read the JSON and to do the calls and that is how we faced this size limit (?) issue.

**Comment by agpmilli** [comment]
Anyone else facing this issue? Any workaround?

Thanks in advance.

**Comment by Wend4r** [comment]
Over HTTPS and RiP on Linux, Response.Data works correctly for me with more than 86554 characters

**Comment by agpmilli** [comment]
Thanks.
We are also using HTTPS but our test server is on Windows. This might be the issue. We will give it a try with RiP on Linux.

In the callback, we are doing: `JSONObject data = view_as<JSONObject>(response.Data);` and this is the line returning the error. Should we do it differently?

**Comment by Wend4r** [comment]
I do
```sourcepawn
			JSON hResponseData = bResponseIsSuccessful ? hResponse.Data : null;
```

**Comment by agpmilli** [comment]
Thanks for the input. It actually works perfectly well on Linux. 

Seems like the issue is with RiP on Windows. 
Anything we can do to make it work on Windows as well?

**Comment by ErikMinekus** [comment]
Try the debug build posted here: https://forums.alliedmods.net/showthread.php?p=2749617#post2749617

And post your server console output here.

**Comment by agpmilli** [comment]
It seems to work with this build. I receive a 200 and it does not crash at:
```
JSONObject data = view_as<JSONObject>(response.Data);
```

Here is what I see in the server console:
[api_ripext.txt](https://github.com/ErikMinekus/sm-ripext/files/6950540/api_ripext.txt)

**Comment by ErikMinekus** [comment]
> Is it possible that this issue is fixed in this debug release and not in the latest windows release?

No, the only difference between the 1.3.0 release and debug builds is that it prints those debug messages.

**Comment by agpmilli** [comment]
Can this be due to the fact that in the non-debug windows build the stream is not finished being read (as we don't print it) and it gets returned as an incomplete body that then fails to be JSONified?

**Comment by ErikMinekus** [comment]
It shouldn't, libcurl is supposed to read the response in 16384 byte chunks, and once the complete response is loaded into memory, it will call the callback. And as you saw, that works fine on Linux. So something must be going wrong in libcurl on Windows where it thinks the response has been completely received.

**Comment by agpmilli** [comment]
Ok. So, nothing we can do here?
This is still weird that similar builds are not behaving the same.

For development purpose, I will keep this debug build (that works on Windows) and when deployed (on Linux machines), the Linux non-debug build seems to work.

**Comment by rtldg** [comment]
Do you know if there's a curl issue/ticket regarding the Windows issue?
